server/mongo_indexes.go

```diff
@@ -21,9 +22,13 @@ type IndexMap map[string][]*mgo.Index
 // on the size of the collection it may take some time before the index is created.
 // This will block the current thread until the indexing completes, but will not block
 // other connections to the mongo database.
-func ConfigureIndexes(session *mgo.Session, config Config) {
+func ConfigureIndexes(connection *MongoConnection, config Config) {
 	var err error
-	db := session.DB(config.DatabaseName)
+
+	session := connection.Copy()
+	session.SetTimeout(5 * time.Minute) // Some indexes take a long time to build
+	defer session.Close()
```

Comment:
I can't imagine there being a problem with `session.SetTimeout`, but if there was, then this session would never get closed.  Ideally, `defer session.Close()` should be immediately after copying it.